viam_cartographer.go

```diff
@@ -122,21 +122,35 @@ func TerminateCartoLib() error {
 func initSensorProcess(cancelCtx context.Context, cartoSvc *CartographerService) {
 	spConfig := sensorprocess.Config{
 		CartoFacade:       cartoSvc.cartofacade,
-		Lidar:             cartoSvc.lidar.testing,
+		Lidar:             cartoSvc.lidar.timed,
 		LidarName:         cartoSvc.lidar.name,
 		LidarDataRateMsec: cartoSvc.lidar.dataRateMsec,
+		IMU:               cartoSvc.imu.timed,
+		IMUName:           cartoSvc.imu.name,
+		IMUDataRateMsec:   cartoSvc.imu.dataRateMsec,
 		Timeout:           cartoSvc.cartoFacadeTimeout,
 		Logger:            cartoSvc.logger,
 	}
 
 	cartoSvc.sensorProcessWorkers.Add(1)
 	go func() {
 		defer cartoSvc.sensorProcessWorkers.Done()
-		if jobDone := sensorprocess.Start(cancelCtx, spConfig); jobDone {
+		if jobDone := sensorprocess.StartLidar(cancelCtx, spConfig); jobDone {
 			cartoSvc.jobDone.Store(true)
 			cartoSvc.cancelSensorProcessFunc()
 		}
 	}()
+
+	if spConfig.IMUName != "" {
```

Comment:
What happens if IMUName is non empty but `config_params.use_imu_data` is true? or vice versa?

Reply:
I am slightly confused - if IMU name is nonempty, then we want to be using emu data, so the variable should be true, right?
If you are asking about the case in which the variable is set to true, but we do not have an IMU configured, or we have an IMU configured, but the variable is set to false - this is addressed in my upcoming commit, see lines 440-445 of viam_cartographer.go

Reply:
Wouldn't it be good to log some sort of indication that it is being configured to use an IMU vs not?